README.md

```diff
@@ -259,6 +259,94 @@ If you lose these details at any stage you can always reauthorise by redirecting
 
 Once you are approved as a Xero Partner you can request unofficial documentation to do with customizing Payment Services and Branding Themes using the API. There is more info on that [here](  https://github.com/waynerobinson/xeroizer/pull/439).
 
+### OAuth2 Applications
+
+For more details, checkout Xero's [documentation](https://developer.xero.com/documentation/oauth2/auth-flow) and [migration steps](https://developer.xero.com/documentation/oauth2/migrate).
+
+1. Generate the authorization url and redirect the user to authenticate
+```ruby
+client = Xeroizer::OAuth2Application.new(
+	YOUR_OAUTH2_CLIENT_ID,
+	YOUR_OAUTH2_CLIENT_SECRET,
+)
+url = client.authorize_url(
+	# The URL's domain must match that listed for your application
+	# otherwise the user will see an invalid redirect_uri error
+	redirect_uri: YOUR_CALLBACK_URL,
+	# space separated, see all scopes at https://developer.xero.com/documentation/oauth2/scopes.
+	# note that `offline_access` is required to get a refresh token, otherwise the access only lasts for 30 mins and cannot be refreshed.
+	scope: "accounting.settings.read offline_access"
+)
+
+# Rails as an example
+redirect_to url
+```
+
+2. In the callback route, use the provided code to retrieve an access token.
+
+```ruby
+token = client.authorize_from_code(
+	params[:code],
+	redirect_uri: YOUR_CALLBACK_URL,
+	grant_type: 'authorization_code'
+)
```

Comment:
This didn't work for me, so I had to use a different approach:
```ruby
enc_client = Base64.strict_encode64("#{YOUR_OAUTH2_CLIENT_ID}:#{YOUR_OAUTH2_CLIENT_SECRET}")
token = client.authorize_from_code(
  params[:code],
  redirect_uri: YOUR_CALLBACK_URL,
  headers: { 'Authorization' => "Basic #{enc_client}" }
)
```
I used as model the way how's done in [xero-ruby](https://github.com/XeroAPI/xero-ruby/blob/23389d3d89c3b44757e4089ccc5b0a1ab1943efc/lib/xero-ruby/api_client.rb#L135). Also, `grant_type` wasn't required as it's [already implicitly added by oauth2 gem](https://github.com/oauth-xx/oauth2/blob/ebe95171c3d34b9f8f4a4dfdca2e5094efc2deee/lib/oauth2/strategy/auth_code.rb#L31) later.

Reply:
To clarify, the `code` here is what you received in the callback from Xero? The parallel for this step is actually https://github.com/XeroAPI/xero-ruby/blob/master/lib/xero-ruby/api_client.rb#L109, (can see it referenced in the [README](https://github.com/XeroAPI/xero-ruby#user-authorization--callback) as well), which doesn't include this header 🤔 

Thanks for catching the `grant_type`! TIL

Reply:
I've just checked without the `headers` parameter and it's working.
It's my fault for not looking into the PRs for any updated readme. I figured out by myself how to connect with this new OAuth2 approach, using xero-ruby as model about what was sent to the API and after I finished that I found your PR with all the info 🤦‍♂️ 
I suppose I had issues setting up the `redirect_uri` yesterday and I thought it was because of that missing header that xero-ruby includes, but the id and secret are already included as part of the body... anyway, disregard what I said about `headers`.

Reply:
All good! I had to dig around the code to figure out how stuff worked, so it's very possible I made mistakes / missed easier ways of doing things (e.g. I didn't notice `current_connections` initially). Appreciate your pointers!